docs/FAQ.md

```diff
@@ -298,14 +298,180 @@ jj config set --user git.private-commits "'''description(glob:'private:*')'''"
+working on "featureB" without changing revisions:
+
+```console
+$ jj log
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+◆  zzzzzzzz root() 00000000
+$ cat file  # Oh no, "feature B" does not belong in this revision...
+Done with feature A
+Working on feature B
+```
+
+#### Step 1: Find the commit id for the "last good version"
+
+<!-- TODO: Reorganize the two related questions, this one and
+  -- the one linked below
+  -->
+You can find [all the past versions of the working copy revision that `jj` has
+saved](#jj-is-said-to-record-the-working-copy-after-jj-log-and-every-other-command-where-can-i-see-these-automatic-saves)
+by running `jj evolog`. The obsolete versions will be marked as "hidden" and
+will have the same change id, but will have different commit ids. For example,
+this is what the evolog might look like after you made two edits to the same
+commit:
+
+```console
+$ # Note the "hidden", the commit ids on the right, and the unchanging change id on the left
+$ jj evolog
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 21:00:51 b8004ab8
+│  featureA
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 20:50:05 e4d831d
+   (no description set)
+```
+
+Since commit `b800` is hidden, it is considered obsolete and `jj log` (without
+arguments) will not show it, nor can it be accessed by its change id. However,
+most `jj` operations will work on it if you refer to it by its commit id.
+
+To find out which of these versions is the last time before we started working
+on feature B (the point where we should have created a new revision, but failed
+to do so), we can look at the actual changes between the `evolog` commits by
+running `jj evolog -p`:
+
+```console
+$ # When was the last saved point before we started working on feature B?
+$ jj evolog -p --git  # We use `--git` to make diffs clear without colors
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+│  diff --git a/file b/file
+│  index 2b455c4207..2a7e05a01a 100644
+│  --- a/file
+│  +++ b/file
+│  @@ -1,1 +1,2 @@
+│   Done with feature A
+│  +Working on feature B
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 21:00:51 b8004ab8
+│  featureA
+│  diff --git a/file b/file
+│  index cb61245109..2b455c4207
+│  --- a/file
+│  +++ b/file
+│  @@ -1,1 +1,1 @@
+│  -Working on feature A
+│  +Done with feature A
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 20:50:05 e4d831d
+   (no description set)
+   diff --git a/file b/file
+   index 0000000000..cb61245109
+   --- /dev/null
+   +++ b/file
+   @@ 0,0 +1,1 @@
+   +Working on feature A
+...
+```
+
+The version of the revision when we were actually done with "featureA" is thus
+saved in the commit with id `b800`. The change after that belongs in a new
+revision.
+
+#### Step 2: Restore the existing revision to an older state and create a new revision with the current state
```

Comment:
This sounds backwards, especially when you explain with "**First**, create a new empty child commit".

Reply:
Good point, I think I was thinking about it backwards.